### PR TITLE
updated model l3_interfaces IOS

### PR DESCRIPTION
**Pull request by KB-perByte**
updated model for l3 interfaces with added attributes and deprecated options

### CODE REVIEW

File: models/ios/l3_interfaces/ios_l3_interface.yaml [hunk 1]
```diff
@@ -1,88 +1,136 @@
...
+                description: Insert default route.
+                type: bool
+          dhcp:
+            description: Obtain a ipv6 address using DHCP.
+            type: dict
+            suboptions:
+              rapid_commit:
+                description: Enable Rapid-Commit.
+                type: bool
+          ipv6_config:
```
Comment (NilashishC) [inline models/ios/l3_interfaces/ios_l3_interface.yaml:92]: This doesn't seem right to me. If these are booleans, that is how they should be added in the model.

File: models/ios/l3_interfaces/ios_l3_interface.yaml [hunk 2]
```diff
@@ -1,88 +1,136 @@
...
+            description: Other configuration options for ipv6 interfaces
+            type: str
+            choices:
+              - anycast
+              - cga
+              - eui-64
+              - segment-routing
```
Comment (NilashishC) [inline models/ios/l3_interfaces/ios_l3_interface.yaml:99]: segment-routing enters into a separate mode which needs to be handled.

an-ios-02(config-if)#ipv6 address 2001:DB8:0:3::/64 segment-routing ?
  <cr>

an-ios-02(config-if)#ipv6 address 2001:DB8:0:3::/64 segment-routing
%GigabitEthernet0/0: Warning: 2001:DB8:0:3::/64 is a Subnet Router Anycast
an-ios-02(config-if-sr-ipv6)#?
ipv6 address segment-routing mode configuration commands:
  default  Set a command to its defaults
  exit     Exit from SR submode
  ipv6-sr  Request options specific to IPV6 segment-routing
  no       Negate a command or set its defaults

an-ios-02(config-if-sr-ipv6)#

Comment (KB-perByte) [reply models/ios/l3_interfaces/ios_l3_interface.yaml]: @NilashishC Added as per the options, skipped `no` and `exit`

File: models/ios/l3_interfaces/ios_l3_interface.yaml [hunk 3]
```diff
@@ -1,88 +1,169 @@
...
+              enable:
+                description: enable auto-configuration.
+                type: bool
+              default:
+                description: Insert default route.
+                type: bool
+          dhcp:
+            description: 
+            - Obtain a ipv6 address using DHCP.
+            - This option is DEPRECATED and is replaced with dhcp which
```
Comment (NilashishC) [inline models/ios/l3_interfaces/ios_l3_interface.yaml:95]: Is the alternate option mentioned here correct?

File: models/ios/l3_interfaces/ios_l3_interface.yaml [hunk 4]
```diff
@@ -1,88 +1,169 @@
...
+          secondary:
+            description:
+            - Configures the IP address as a secondary address.
+            type: bool
+          dhcp_client:
+            description:
+            - Configures and specifies client-id to use over DHCP ip. Note, This option
+              shall work only when dhcp is configured as IP.
+            - GigabitEthernet interface number
+            - This option is DEPRECATED and is replaced with dhcp which
```
Comment (NilashishC) [inline models/ios/l3_interfaces/ios_l3_interface.yaml:44]: s the alternate option mentioned here correct?

File: models/ios/l3_interfaces/ios_l3_interface.yaml (hunk 3 above)
Comment (KB-perByte) [reply models/ios/l3_interfaces/ios_l3_interface.yaml:95]: https://docs.ansible.com/ansible/latest/collections/cisco/ios/ios_l3_interfaces_module.html#parameter-config/ipv6 I found out that there was no implementation of dhcp under ipv6 so I renamed it back to dhcp as the parent tag. for both ipv4 and ipv6 the pending PR that had a model made me confused.

**Review by NilashishC — APPROVED** [review]

**Pull request merged**